### PR TITLE
simulation-based resubstitution & wrapper function of FRAIG based on 0-insertion resub

### PR DESCRIPTION
**Pull request by lee30sonia**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,5 @@ examples/
 
 docs/_build
 docs/doxyxml
+
```
Comment (msoeken) [inline .gitignore]: I think this went here by mistake?

File: include/mockturtle/algorithms/fraig.hpp [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  ps.verbose = psf.verbose? psf.verbose: false;
+  ps.max_pis = psf.max_pis? psf.max_pis: 10u;
+  ps.max_inserts = 0u;
+  ps.random_seed = psf.random_seed? psf.random_seed: 0;
+  
+
+  sim_resubstitution( ntk, ps, &st );
+  ntk = cleanup_dangling( ntk );
+
+  if ( pst )
```
Comment (msoeken) [inline include/mockturtle/algorithms/fraig.hpp]: Doesn't `*pst = st` work?

File: include/mockturtle/algorithms/sim_resub.hpp [hunk 3]
```diff
@@ -0,0 +1,1019 @@
...
+/*!
+  \file sim_resub.hpp
+  \brief Resubstitution
+
+  \author Siang-Yun Lee
+*/
+
+#pragma once
+
+#include <mockturtle/networks/aig.hpp>
```
Comment (msoeken) [inline include/mockturtle/algorithms/sim_resub.hpp]: mockturtle headers should be included with quotes.

File: include/mockturtle/algorithms/sim_resub.hpp [hunk 4]
```diff
@@ -0,0 +1,1019 @@
...
+
+  using resub_view_t = fanout_view2<depth_view<Ntk>>;
+  depth_view<Ntk> depth_view{ntk};
+  resub_view_t resub_view{depth_view};
+
+  simresub_stats st;
+
+  detail::simresub_impl<Ntk, resub_view_t> p( ntk, resub_view, ps, st );
+  p.run();
+  if ( ps.verbose )
```
Comment (msoeken) [inline include/mockturtle/algorithms/sim_resub.hpp]: Consider using `{ ... }` also for single statement blocks.

File: include/mockturtle/algorithms/fraig.hpp (hunk 2 above)
Comment (lee30sonia) [reply include/mockturtle/algorithms/fraig.hpp]: I don't know; fraig_stats and simresub_stats are different

Comment (msoeken) [reply include/mockturtle/algorithms/fraig.hpp]: Okay, makes sense.

**Pull request closed without merging**